projects/packages/publicize/src/social-image-generator/class-extractor.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * Extractor class.
+ *
+ * @package automattic/jetpack-publicize
+ */
+
+namespace Automattic\Jetpack\Publicize\Social_Image_Generator;
+
+/**
+ * Class for extracting SIG-related settings for a post.
+ */
+class Extractor {
+	/**
+	 * Post to get information from.
+	 *
+	 * @var int $post_id
+	 */
+	public $post_id;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param int $post_id Post to get information from.
+	 */
+	public function __construct( $post_id ) {
+		$this->post_id = $post_id;
+	}
+
+	/**
+	 * Get the SIG settings for a post.
+	 *
+	 * @return array
+	 */
+	private function get_post_settings() {
+		$social_options = get_post_meta( $this->post_id, \Automattic\Jetpack\Publicize\Publicize::POST_JETPACK_SOCIAL_OPTIONS, true );
+
+		if ( ! is_array( $social_options ) || empty( $social_options['image_generator_settings'] ) ) {
+			return array();
+		}
+
+		return $social_options['image_generator_settings'];
+	}
+
+	/**
+	 * Check if SIG is enabled for a specific post.
+	 *
+	 * @return bool
+	 */
+	public function is_enabled() {
+		$settings = $this->get_post_settings();
+
+		return ! empty( $settings['is_enabled'] );
```

Comment:
This is called `enabled` now, and as the default is true, the value doesn't get saved if you don't change the toggle, so we might need to make this something like `return $settings['enabled'] ?? true` as long as we're checking that the feature is enabled elsewhere.

Reply:
> This is called `enabled` now

Fixed in https://github.com/Automattic/jetpack/pull/29118/commits/91aa91883c2eba1c349c46b0725beb5e2bf437d4

> we might need to make this something like `return $settings['enabled'] ?? true` as long as we're checking that the feature is enabled elsewhere.

I don't think this would work, because enabling the feature would suddenly enable SIG for all posts that didn't explicitly disable it in the post meta.

What about explicitly setting the `enabled` meta in the `save_post` hook that sets the token? When that hook runs we can be confident that the feature is enabled, and it would be safe to check if the `enabled` value is set to `false`. If it's not, we can add `enabled => true` to the post meta.

Reply:
> I don't think this would work, because enabling the feature would suddenly enable SIG for all posts that didn't explicitly disable it in the post meta.

Good point. Explicitly setting the value seems like the way to go.

Reply:
Done: https://github.com/Automattic/jetpack/pull/29118/commits/22d05ae7198b7e61ad1267687b52462ed1322987